examples/run/cuda/inst_nvtx.cpp

```diff
@@ -0,0 +1,88 @@
+/* Copyright (c) 2014, NVIDIA CORPORATION. All rights reserved.
```

Comment:
Please explain this file. I can't imagine that any recommendation from NVIDIA would include user code having to carry a copy of such a source file with itself.

I get the sense that you "just" wanted to provide these `rangePush(...)` and `rangePop()` functions to the user code. But you didn't add calls to them anywhere. So I can't imagine that this source file, in its current form, would be doing anything. :confused:

Reply:
The `__cyg_profile_func_enter` and `__cyg_profile_func_exit` functions are called automatically when function instrumentation is enabled. :wink:

Reply:
Okay, code-wise it makes some sense at least then. But there are just multiple things in it that I don't like. The non-static global variable declarations being the very first...